### DISCUSSION TITLE
[ENG-178] Wildcard match for infisical.json

### DISCUSSION THREAD

**Discussion by imrane** [Infisical/infisical#5713] [Feature Request]
### Feature description

Right now the branch and related secrets are hardcoded...ie.

```
{
  "workspaceId": "63ee5410a45f7a1ed39ba118",
  "gitBranchToEnvironmentMapping": {
    "branchName": "dev",
    "anotherBranchName": "staging"
  }
}
```

Would be more useful to have wildcard matching similar to how secrets are linked, so we could write something like...

```
{
  "workspaceId": "63ee5410a45f7a1ed39ba118",
  "gitBranchToEnvironmentMapping": {
    "feat/**": "dev",
    "fix/**": "dev"
  }
}
```

### Why would it be useful?

Many teams have multiple active branches in development and name them accordingly e.g.`feat/add-infisical-support`. A wildcard would better match these branches.

### Additional context

I think it's fairly straight forward.

<sub>From [SyncLinear.com](https://synclinear.com) | [ENG-178](https://linear.app/infisical/issue/ENG-178/wildcard-match-for-infisicaljson)</sub>

**Comment by saikumarpb** [comment]
@vmatsiiako Can i work on this if this is open to be picked up ?

**Comment by jorgeecardona** [comment]
Any progress on this?
